### DISCUSSION TITLE
Delegates

### DISCUSSION THREAD

**Discussion by danp11** [Randgalt/record-builder#77] [Ideas]
Hi all, 

As records favor composition, I wanted to discuss with you if an additional interface, for example 
`public interface Delegates extends With` 
could be something that would make sence?

Let say we have a record 
`public record Trade(TradeInfo tradeInfo) implements TradeBuilder.With`

where TradeInfo looks like this:
`public record TradeInfo(String orderbookId, String tradeId) {}`

this delegate interface could look something like this

```
public interface Delegates extends With {

        default String orderbookId() {
            return tradeInfo().orderbookId();
        }

        default String tradeId() {
            return tradeInfo().tradeId();
        }
    }
```

This would let us specify trade.tradeId instead of trade.tradeInfo.tradeId when we are using the trade record.

What are your thoughts about this delegate interface?

Regards
Dan

**Comment by Randgalt** [comment]
This seems like a lot of work just to write `trade.tradeId()` instead of `trade().tradeInfo().tradeId()`. I'm -1 on this. Also, this isn't in the category of "building records". Maybe others will have some ideas on this.

**Reply by danp11** [reply]
Hi, I understand that it is out of scope for this project. 

It's just when modelling large modells with record blocks the delegates would save alot of "extra getter code"

You can close this discussion.
/Dan

**Comment by Randgalt** [comment]
We can leave it open - maybe others have more ideas
